### ISSUE TITLE
Research graph and data visualization libraries

### ISSUE THREAD

**Issue by txau** [huridocs/uwazi#1057]
@RafaPolit @daneryl @konzz Lets define some criteria and come to a short list of candidates.

@danicatalan will base his designs in the developer's choice.

**Comment by konzz** [comment]
For charts http://c3js.org/

**Comment by RafaPolit** [comment]
Really? Why? They look so 'thin' and lacking all character to me.  Are we set on this? Is this your preference? Can we elaborate on the reason instead of just the name so we can have an informed decision by the end of the week?

**Comment by txau** [comment]
@RafaPolit this is a discussion, we are not set in anything. Please post your suggestions.

**Comment by danicatalan** [comment]
@RafaPolit the C3.js is based on D3.js library. All is is rendered on SVG, so it's easy to manipulate with CSS properties, so in case you decide this is the best library to implement is not a problem for my to manipulate the styles and give it our style.

![image](https://user-images.githubusercontent.com/6364153/27144437-21c38718-5132-11e7-9e69-e7a3145dae56.png)

**Comment by danicatalan** [comment]
- [ ] Test libraries on mobile.
- [ ] Check library weights.
- [ ] Review C3.js react wrapper.

**Comment by danicatalan** [comment]
@RafaPolit @konzz @daneryl how this is going? do we have any conclusion about the research?

**Comment by RafaPolit** [comment]
Well, we implemented all the different options, and we are down to two:
- C3.js
- Recharts

The first one: C3.js is a fully fledged and very robust option, but options to personalize the way it is displayed is not that great.  It is also a bit 'heavy', with about 400Kb.  On the other hand, user interaction is fully developed. Biggest downside is not being developed in React, which no only requires a container (not that hard) but interaction with it uses a different paradigm than what we have in our tech stack.

The second one: Recharts is lighter, with great potential, still a bit 'young'.  Options to personalize the looks are almost endless, and since it is native React, we can actually do with it whatever we want.  The problem is that user interaction is less developed and we need to develop that part.  Good news is that I have managed to 'replicate' some part of C3's functionality with a days work and a bit of trial and error.  Once we are more familiar with it, I'm sure things will come much easier.

So I'm inclined to go with the second option.  Still, I hesitate about not picking the 'big contender'.  Given our track record of having opted for the wrong database, the wrong PDF approach, and sometimes even wonder about if redux was the way to go, I am not that confident of calling the final decision on Recharts.  Wouldn't want to get half-way down the road and decide to revert to C3.js.
